presto-product-tests/src/main/java/io/prestosql/tests/hive/HiveVersionProvider.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.tests.hive;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Module;
+import io.prestosql.tempto.configuration.Configuration;
+import io.prestosql.tempto.initialization.SuiteModuleProvider;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import java.sql.DatabaseMetaData;
+import java.sql.SQLException;
+import java.util.Optional;
+
+import static io.prestosql.tests.utils.QueryExecutors.onHive;
+
+public class HiveVersionProvider
+{
+    private Optional<HiveVersion> hiveVersion = Optional.empty();
+
+    @Inject
+    public HiveVersionProvider() {}
+
+    public HiveVersion getHiveVersion()
+    {
+        if (hiveVersion.isEmpty()) {
+            detectHiveVersion();
+        }
+        return hiveVersion.get();
+    }
+
+    private void detectHiveVersion()
```

Comment:
make this method return `HiveVersion` (then may be static)
this improves flow in `getHiveVersion()`